doc/sphinx/language/gallina-extensions.rst

```diff
@@ -966,12 +954,35 @@ are now available through the dot notation.
 .. exn:: Signature components for label @ident do not match.
     :undocumented:
 
-.. exn:: This is not the last opened module.
-    :undocumented:
-
-.. exn:: This is not the last opened module type.
+.. exn:: The field @ident is missing in @qualid.
    :undocumented:
 
+.. |br| raw:: html
+
+    <br>
+
+.. note::
+
+  #. Interactive modules and module types can be nested.
+  #. Interactive modules and module types can't be defined inside of :ref:`sections<section-mechanism>`.
+     Sections can be defined inside of interactive modules and module types.
+  #. Hints and notations (:cmd:`Hint` and :cmd:`Notation` commands) can also appear inside interactive
+     modules and module types. Note that with module definitions like:
+
+     :n:`Module @ident__1 : @module_type := @ident__2.`
+
+     or
+
+     :n:`Module @ident__1 : @module_type.` |br|
+     :n:`Include @ident__2.` |br|
+     :n:`End @ident__1.`
+
+     hints and the like valid for :n:`@ident__1` are the ones defined in :n:`@module_type`
+     rather then those defined in :n:`@ident__2` (or the module body).
+
+  #. Within an interactive module type definition, the :cmd:`Parameter` command declares a
+     constant instead of definining a new axiom (which it does when not in a module type definition).
+
 .. cmd:: Include @module_type_inl {* <+ @module_expr_inl }
```

Comment:
Since in the syntax of `Module` and `Module Type` use `{+<+ @module_expr_inl}`, I think it would make sense that the `Include` command uses the same, i.e.
```suggestion
.. cmd:: Include {+<+ @module_expr_inl }
```

Reply:
Really?? module_type_inl and module_type_expr are not the same thing.  I will do it for `Include Type`, though.

Reply:
I hadn't realized these were not the same non-terminals. I wonder why it is so, given the description.